src/dune_lang/lib_kind.mli

```diff
@@ -13,11 +13,33 @@ module Ppx_args : sig
   val empty : t
 end
 
+module Dune_file : sig
+  (** Representation of of valid dune lang values for the library [kind]
+      field *)
+  type t =
+    | Normal
+    | Ppx_deriver of Ppx_args.t
+    | Ppx_rewriter of Ppx_args.t
+
+  (** [cstr_name k] is the name of the kind constructor for [k] in dune lang.
+      E.g., [cstr_name Normal = "normal"] *)
+  val cstr_name : t -> string
+
+  val equal : t -> t -> bool
```

Comment:
Is this used anywhere? If not, let's get rid of it as the implementation relies on porymorphic equality